configs/swarm/docker-compose.local.yml

```diff
@@ -64,7 +65,10 @@ services:
       P2P_ADDRESS: 0.0.0.0:1338
       API_ADDRESS: 0.0.0.0:8081
       GENESIS_PUBLIC_KEY: ed01204164BF554923ECE1FD412D241036D863A6AE430476C898248B8237D77534CFC4
+      GENESIS_PRIVATE_KEY: 80264082B3BDE54AEBECA4146257DA0DE8D59D8E46D5FE34887DCD8072866792FCB3AD4164BF554923ECE1FD412D241036D863A6AE430476C898248B8237D77534CFC4
```

Comment:
why is this required?

both `GENESIS_PUBLIC_KEY` and `GENESIS_PRIVATE_KEY` should be removed from docker-compose files. Instead, if this is required, nodes should pick it from the authority of genesis transactions

Reply:
but preferably, `kagami` can put a random public key as genesis account id. Then peers just take that from genesis transaction authority

Reply:
> Instead, if this is required, nodes should pick it from the authority of genesis transactions

but preferably peers should not care about which account is the authority of genesis transactions

Reply:
In `genesis.json` there can be genesis account id, in e.g. `Transfer` instructions. So do I understand correctly that you propose that `kagami` will generate new genesis account id and replace existing genesis account id in `genesis.json` with generated one?

Reply:
> In genesis.json there can be genesis account id, in e.g. Transfer instructions. So do I understand correctly that you propose that kagami will generate new genesis account id and replace existing genesis account id in genesis.json with generated one?

this is what I was thinking since I see genesis account as some dummy value that is more like a placeholder for `Transaction::authority`. But since genesis can create entities, those entities need to have some owner as you point out. How about we define a field `genesis` in `genesis.json` that holds `AccountId` of genesis account?

Reply:
it's not important that transactions are signed by this genesis account. Genesis transactions don't even have to be signed

Reply:
> it's not important that transactions are signed by this genesis account. Genesis transactions don't even have to be signed

Yes but in current data model `SignedTransaction` must have some `signature`, if it is not genesis account signature, then how peers will know what is genesis account id?

I think users may use `genesis@genesis` as alias for genesis account id in `genesis.json`, and `kagami` will generate genesis key pair, use it as authority for genesis transaction, so in this case users will not need to know anything about genesis key pair. But `genesis@genesis` can't be used as account id, so another option I think of is that users use some genesis account id in `genesis.json`, and then `kagami` generates new genesis account id and replace with it in `genesis.json`

Reply:
> Yes but in current data model SignedTransaction must have some signature, if it is not genesis account signature, then how peers will know what is genesis account id?

from `Transaction::authority`

Reply:
we could also consider introducing a separate type `GenesisBlock` in `data_model` if we can't make it as yet another `SignedBlock`

Reply:
introduction of `GenesisBlock` as separate type would make block sync more complex because now there is 2 types of block we want to sync.

Reply:
> > Yes but in current data model SignedTransaction must have some signature, if it is not genesis account signature, then how peers will know what is genesis account id?
> 
> from `Transaction::authority`, and `Transaction::authority` should be taken from `genesis` field in `genesis.json`. Genesis private key will never be needed by any of the peers

So should it work like this:
* Add `genesis` field to `genesis.json` with genesis `AccountId`
* `kagami` conversion from `genesis.json` to `SignedBlock`:
  * Remove genesis key pair arguments
  * Genesis transactions authority will be genesis account specified in `genesis.json`
  * Genesis transactions will be signed by some random key pair
  * Match of transaction `authority` and `signatory` will not be checked for genesis transactions

Reply:
> introduction of `GenesisBlock` as separate type would make block sync more complex because now there is 2 types of block we want to sync

yes, but different block types will never interchange. So you can think of block sync operating in 2 phases which could be modeled as a state machine. First there is genesis phase and only after it do all other blocks come.

Reply:
Changed implementation according to [this comment](https://github.com/hyperledger/iroha/pull/4793#discussion_r1661199227) (PR description also updated). Please re-review